### PR TITLE
Clarify how to use a negated expression in an if statement

### LINKED ISSUE #28308: Clarify how to use a negated expression in an if statement
### Code of Conduct

- [x] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/learn-github-actions/expressions

### What part(s) of the article would you like to see updated?

The documentation for `if` statements needs checked.

We should clarify how to write an if statement with a negated expression.

### Additional information

I've opened this issue as a result of @fransf-wtax's comment on this already-closed issue:

https://github.com/github/docs/issues/3001#issuecomment-1702464558

> This seems to have regressed as data/reusables/actions/expression-syntax-if.md no longer contains the changes made for this issue. To make matters worse, the current text seems to discourage surrounding expressions in ${{ and }} since they are then supposedly treated as a string; but no solution is given for the scenario described in this ticket where the expression starts with !.

See my comment on that issue that gives the context of what we currently have in the docs.

### Content design plan

- [ ] In the "About expressions" section of the Expressions article (https://docs.github.com/en/actions/learn-github-actions/expressions#about-expressions):

  - [ ] Remove `{% data reusables.actions.expression-syntax-if %}` - see the Markdown file [here](https://github.com/github/docs/blob/main/content/actions/learn-github-actions/expressions.md?plain=1#L26C1-L26C50).

  - [ ] Move the "For more information..." from that line 👆 into a note box:

    ```
    {% note %}

    **Note**: The exception to this rule is when you are using expressions in an `if` clause, where, optionally, you can usually omit {% raw %}`${{`{% endraw %} and {% raw %}`}}`{% endraw %}. For more information about `if` conditionals, see "[AUTOTITLE](/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif)."

    {% endnote %}
    ```

  - [ ] Remove the `if` example [here](https://github.com/github/docs/blob/main/content/actions/learn-github-actions/expressions.md?plain=1#L30-L36).

- [ ] In the `expression-syntax-if` reusable https://github.com/github/docs/blob/main/data/reusables/actions/expression-syntax-if.md?plain=1:

  - [ ] In the first paragraph, change "you may omit" to "you can, optionally, omit".

  - [ ] In the first paragraph, change "However, this rule does not" to "However, this exception does not".

  - [ ] In the second paragraph, change "You must use the" to "You must always use the".

  - [ ] At the end of the second paragraph, add: "For example:" followed by:

    ````
    [blank line]
    ```yaml
    if: ${{ ! startsWith(github.ref, 'refs/tags/') }}
    ```
    [blank line]
    ````

  - [ ] Delete the third paragraph, beginning "Using the..."

### PR DESCRIPTION
**Pull request by albbus-stack**
### Why:

Makes a clearer explanation of a negated expression inside an if statement.

Closes: #28308

### What's being changed (if available, include any code snippets, screenshots, or gifs):

All the changes are as requested by @hubwriter in [this comment](https://github.com/github/docs/issues/28308#issuecomment-1735267416).

### Check off the following:

- [ ] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by albbus-stack** [comment]
> @albbus-stack Thanks so much for submitting a PR! I'll get this triaged for review ⚡

The code block in `data/reusables/actions/expression-syntax-if.md` fails the `content-linter` action with:

```bash
Error name:    "ParseError"
Error message: "unexpected token at \"! startsWith(...\", line:6, col:6"
```

File: data/reusables/actions/expression-syntax-if.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
...
+When you use expressions in an `if` conditional, you can, optionally, omit the {% raw %}`${{ }}`{% endraw %} expression syntax because {% data variables.product.prodname_actions %} automatically evaluates the `if` conditional as an expression. However, this exception does not apply everywhere.
 
-You must use the {% raw %}`${{ }}`{% endraw %} expression syntax or escape with `''`, `""`,  or `()` when the expression starts with `!`, since `!` is reserved notation in YAML format.
+You must always use the {% raw %}`${{ }}`{% endraw %} expression syntax or escape with `''`, `""`,  or `()` when the expression starts with `!`, since `!` is reserved notation in YAML format. For example:
 
-Using the {% raw %}`${{ }}`{% endraw %} expression syntax turns the contents into a string, and strings are truthy. For example, `if: true && {% raw %}${{ false }}{% endraw %}` will evaluate to `true`.
+```yaml
+if: ${{ ! startsWith(github.ref, 'refs/tags/') }}
```
Comment (SiaraMist) [inline data/reusables/actions/expression-syntax-if.md:6]:
```suggestion
if: {% raw %}}${{ ! startsWith(github.ref, 'refs/tags/') }}{% endraw %}
```
I'm going to see if this fixes the linter error.

File: data/reusables/actions/expression-syntax-if.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
...
+When you use expressions in an `if` conditional, you can, optionally, omit the {% raw %}`${{ }}`{% endraw %} expression syntax because {% data variables.product.prodname_actions %} automatically evaluates the `if` conditional as an expression. However, this exception does not apply everywhere.
 
-You must use the {% raw %}`${{ }}`{% endraw %} expression syntax or escape with `''`, `""`,  or `()` when the expression starts with `!`, since `!` is reserved notation in YAML format.
+You must always use the {% raw %}`${{ }}`{% endraw %} expression syntax or escape with `''`, `""`,  or `()` when the expression starts with `!`, since `!` is reserved notation in YAML format. For example:
 
-Using the {% raw %}`${{ }}`{% endraw %} expression syntax turns the contents into a string, and strings are truthy. For example, `if: true && {% raw %}${{ false }}{% endraw %}` will evaluate to `true`.
+```yaml
+if: {% raw %}}${{ ! startsWith(github.ref, 'refs/tags/') }}{% endraw %}
```
Comment (SiaraMist) [inline data/reusables/actions/expression-syntax-if.md:6]:
```suggestion
if: {% raw %}${{ ! startsWith(github.ref, 'refs/tags/') }}{% endraw %}
```

File: data/reusables/actions/expression-syntax-if.md [hunk 3]
```diff
@@ -1,5 +1,7 @@
...
+When you use expressions in an `if` conditional, you can, optionally, omit the {% raw %}`${{ }}`{% endraw %} expression syntax because {% data variables.product.prodname_actions %} automatically evaluates the `if` conditional as an expression. However, this exception does not apply everywhere.
 
-You must use the {% raw %}`${{ }}`{% endraw %} expression syntax or escape with `''`, `""`,  or `()` when the expression starts with `!`, since `!` is reserved notation in YAML format.
+You must always use the {% raw %}`${{ }}`{% endraw %} expression syntax or escape with `''`, `""`,  or `()` when the expression starts with `!`, since `!` is reserved notation in YAML format. For example:
 
-Using the {% raw %}`${{ }}`{% endraw %} expression syntax turns the contents into a string, and strings are truthy. For example, `if: true && {% raw %}${{ false }}{% endraw %}` will evaluate to `true`.
+```yaml
+if: {% raw %}${{ ! startsWith(github.ref, 'refs/tags/') }}{% endraw %}
+```
```
Comment (SiaraMist) [inline data/reusables/actions/expression-syntax-if.md:7]:
```suggestion
```

```
I think this needs a blank line at the end.

Comment (SiaraMist) [inline data/reusables/actions/expression-syntax-if.md:7]:
```suggestion
{% raw %}

```yaml
if: ${{ ! startsWith(github.ref, 'refs/tags/') }}
```

{% endraw %}
```

**Review by SiaraMist — APPROVED** [review]
Thank you so much for picking up this issue and making these changes, @albbus-stack! I made a couple of edits just to formatting that got all of the tests passing, so I'll go ahead and get this merged. Thank you again for contributing to the docs! 🚀

**Pull request merged**